boilerPlate/src/app/shared/services/custom-validation.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+} from '@angular/forms';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CustomValidationService {
+  constructor() {}
+
+  passwordValidator: ValidatorFn = (
+    control: AbstractControl
+  ): ValidationErrors | null => {
+    if (control.value === '' || control.value === null) {
+      return null;
+    }
+
+    if (
+      control.value &&
+      control.value.match(
+        new RegExp(/^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,16}$/)
```

Comment:
We can set this Regex in separate constant file under object like 
export const COMMON = {
  REGEX: {
   PASSWORD: new RegExp(/^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,16}$/)
  }
}

Reply:
done